src/Elmish/Component.purs

```diff
@@ -274,20 +274,29 @@ nat map def =
 -- | proven to be fragile in some specific circumstances (e.g. multiple events
 -- | occurring within the same JS synchronous frame), so it is not recommended
 -- | to use this mechanism for complex components.
--- |
--- | Note 3: this function has to be called exactly once, at top-level. It
--- | cannot be called in the parent's `view` function on every render. Every
--- | time it is called it creates a new React class, so if it was called on
--- | every render, the resulting JSX DOM will contain a new component every
--- | time, which will make React remount that component and destroy its local
--- | state. See explanation on `withCachedComponent` for more.
```

Comment:
Is this no longer true?

Reply:
Yes, it hasn't been since the introduction of `ComponentName`, but I forgot to remove the comment at that time.

The original implementation didn't take a `ComponentName` and was indeed creating a whole new component (i.e. new `class Foo extends React.Component`) every time, which led to repeated `componentDidMount` calls and other interesting and tricky issues. I really wanted a clean API.

Then I realized that my struggle for a clean API actually created a very ugly one: this unwritten requirement that you have to call this function only once. Plus, there was a pure technical issue: genericity. 

```
foo :: forall a. Ord a => a -> ReactElement
foo = wrapWithLocalState \a -> ...
```

Here 👆, even though, on the surface, `wrapWithLocalState` is called exactly once, it's really not. `Ord a` is another parameter at JavaScript level, so the function ends up being called on every render.

So I introduced `ComponentName`, which now allows us to be very strict about when to create a new class (in React sense), but at the cost of a slight inconvenience for the consumer.

Reply:
Got it, thanks for the explanation!